app/custom-oauth/server/custom_oauth_server.js

```diff
@@ -344,50 +394,57 @@ export class CustomOAuth {
 			identity.name = this.getName(identity);
 		}
 
-		return identity;
+		// remove all properties with names that don't match the RegExp in `IDENTITY_PROPNAME_FILTER`
+		return Object.entries(identity).reduce((result, [name, value]) => {
+			if (IDENTITY_PROPNAME_FILTER.test(name)) {
+				return result;
```

Comment:
Can you replace the `.` with underscores instead of discarding the attributes? That way we will continue to store the whole data.

Reply:
well, this might get a little tricky due to nested properties. I believe, the odata protocol allows to have multiple levels of props, each of which can have 'invalid' names. Also, there might be arrays that contain objects with such field names.
if you need any value from odata to build the profile, you can use it through the introduces templating mechanism. these fields are only removed after all the templating happened.